query-service-impl/src/main/java/org/hypertrace/core/query/service/prometheus/QueryRequestToPromqlConverter.java

```diff
@@ -0,0 +1,142 @@
+package org.hypertrace.core.query.service.prometheus;
+
+import static org.hypertrace.core.query.service.QueryRequestUtil.getLogicalColumnNameForSimpleColumnExpression;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import org.hypertrace.core.query.service.ExecutionContext;
+import org.hypertrace.core.query.service.QueryRequestUtil;
+import org.hypertrace.core.query.service.QueryTimeRange;
+import org.hypertrace.core.query.service.api.Expression;
+import org.hypertrace.core.query.service.api.Expression.ValueCase;
+import org.hypertrace.core.query.service.api.QueryRequest;
+import org.hypertrace.core.query.service.prometheus.PrometheusViewDefinition.MetricConfig;
+
+class QueryRequestToPromqlConverter {
+
+  private final PrometheusViewDefinition prometheusViewDefinition;
+  private final PrometheusFunctionConverter prometheusFunctionConverter;
+  private final FilterToPromqlConverter filterToPromqlConverter;
+
+  QueryRequestToPromqlConverter(PrometheusViewDefinition prometheusViewDefinition) {
+    this.prometheusViewDefinition = prometheusViewDefinition;
+    this.prometheusFunctionConverter = new PrometheusFunctionConverter();
+    this.filterToPromqlConverter = new FilterToPromqlConverter();
+  }
+
+  PromQLInstantQueries convertToPromqlInstantQuery(
+      ExecutionContext executionContext,
+      QueryRequest request,
+      LinkedHashSet<Expression> allSelections) {
+
+    QueryTimeRange queryTimeRange =
+        executionContext
+            .getQueryTimeRange()
+            .orElseThrow(() -> new RuntimeException("Time Range missing in query"));
+
+    return new PromQLInstantQueries(
+        buildPromqlQueries(
+            request, allSelections, queryTimeRange, executionContext.getTimeFilterColumn()),
+        queryTimeRange.getEndTime());
+  }
+
+  PromQLRangeQueries convertToPromqlRangeQuery(
+      ExecutionContext executionContext,
+      QueryRequest request,
+      LinkedHashSet<Expression> allSelections) {
+
+    QueryTimeRange queryTimeRange =
+        executionContext
+            .getQueryTimeRange()
+            .orElseThrow(() -> new RuntimeException("Time Range missing in query"));
+
+    return new PromQLRangeQueries(
+        buildPromqlQueries(
+            request, allSelections, queryTimeRange, executionContext.getTimeFilterColumn()),
+        queryTimeRange.getStartTime(),
+        queryTimeRange.getEndTime(),
+        getTimeSeriesPeriod(executionContext));
+  }
+
+  private List<String> buildPromqlQueries(
+      QueryRequest request,
+      LinkedHashSet<Expression> allSelections,
+      QueryTimeRange queryTimeRange,
+      String timeFilterColumn) {
+    List<String> groupByList = getGroupByList(request);
```

Comment:
You will have to add tenant filter as default filter? Hopefully, tenant filter is already present in QueryReqeust?

Reply:
yes added tenant filter, updated test